### PR TITLE
Feature: .NET Core 2.0/2.1 debugging support

### PR DESCRIPTION
**Pull request by ndobryanskyy**
*Issue #, if available:*
#500 -> #568

*Description of changes:*

Enabled .NET Core debugging by adding corresponding to lambci/docker-lambda#130 entry point override utilizing `--debugger-spin-wait`.

Also updated corresponding [document](docs/usage.rst) any comments and suggestions are welcome as always.

*Checklist:*

- [x] Write Design Document #759
- [x] Fix unit tests
- [x] Write/update functional tests. Do I need to add any?
- [x] Write/update integration tests. Do I need to add any?
- [x] `make pr` passes
- [x] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by TheSriram** [comment]
Thanks for adding this, its a very exciting feature, with zero changes in SAM CLI's logic itself. 🎆

**Comment by jfuss** [comment]
This is waiting on #823 to finish building so we can fix the build. I will update and merge this in once I have that complete.

File: samcli/local/docker/lambda_container.py [hunk 1]
```diff
@@ -289,8 +297,9 @@ def _get_entry_point(runtime, debug_options=None):
 
     @staticmethod
     def _supported_runtimes():
-        return {Runtime.java8.value, Runtime.go1x.value, Runtime.nodejs.value, Runtime.nodejs43.value,
-                Runtime.nodejs610.value, Runtime.nodejs810.value, Runtime.python27.value, Runtime.python36.value}
+        return {Runtime.java8.value, Runtime.dotnetcore20.value, Runtime.dotnetcore21.value, Runtime.go1x.value,
```
Comment (jfuss) [inline samcli/local/docker/lambda_container.py]: 🎉

**Comment by jfuss** [comment]
@ndobryanskyy We need to wait for the docker-lambda change right?

**Comment by ndobryanskyy** [comment]
@jfuss right. I will get back to you as soon as lambci/docker-lambda#130 is merged.

Thanks for your feedback and help @sanathkr @TheSriram @jfuss I really appreciate that!

I'll fix minor issue with .rst here with unnecessary `` and update the doc PR with our latest findings

**Comment by ndobryanskyy** [comment]
lambci/docker-lambda#130 Merged 🎉

**Pull request merged**